File: cpp/znode_demo/pub.cc
#include <iostream>
#include <unistd.h>
#include <trunk/cater/znode.h>
#include <trunk/msgs/hdmap.pb.h>

using namespace trunk;

int main() {
    std::cout << "Hello, Znode Pub!" << std::endl;
    cater::ZNode znode;
    std::string channel = "ipc:///var/log/trunk/haha.ipc";
    msgs::Position position;
    position.mutable_point()->set_x(1);
    position.mutable_point()->set_y(2);
    position.mutable_point()->set_z(3);
    while (true) {
        sleep(1);
        znode.channel(channel,1, cater::ZNode::PUB).pub("/haha/hdmap", position);
        std::cout << position.DebugString() << std::endl;
    }
    return 0;
}

File: cpp/znode_demo/pub.py
import zmq
import time
from protos.hdmap_pb2 import Position


class Publisher(object):

    def __init__(self, topic):
        self.topic = topic
        self.context = zmq.Context()
        self.socket = self.context.socket(zmq.PUB)
        self.socket.bind("ipc:///var/log/trunk/haha.ipc")
        self.position = Position()
        self.position.point.x = 1
        self.serialize_data = self.position.SerializeToString()

    def run(self):
        topic = bytes(self.topic,encoding = "utf8")
        while True:
            print(self.position)
            self.socket.send_multipart([topic, self.serialize_data])
            time.sleep(1)


if __name__ == '__main__':
    print("pub topic")
    pub = Publisher("/haha/hdmap")
    pub.run()